### ISSUE TITLE
Zod optionals, enums and native enums don't generate proper schema and example

### ISSUE THREAD

**Issue by adsc** [blomqma/next-rest-framework#63]
I have a body schema with following zod config (the parameters for `z.nativeEnum` calls are Prisma enums):
```typescript
z.object({
   species: z.string(),
   reporter: z.string().or(reporterSchema),
   observer: z.string(),
   observerType: z.enum(["Type1", "Type2"]),
   copyright: z.string(),
   dataProtection: z.nativeEnum(DataProtection),
   comments: z.string().optional(),
   x: z.number(),
   y: z.number(),
   xyAccuracy: z.nativeEnum(SpatialAccuracy),
   country: z.nativeEnum(Country),
   canton: z.string(),
   district: z.optional(z.string()),
   municipality: z.string(),
   fieldName: z.string(),
   huntingArea: z.string().optional(),
   sensitive: z.boolean(),
})
```

This is how the example in the swagger doc looks. As you can see, optionals (comments, district, huntingArea) are broken. Enums and native enums are working fine here:
```json
{
  "species": "string",
  "reporter": "string",
  "observer": "string",
  "observerType": "Type1",
  "copyright": "string",
  "dataProtection": "PRIVATE",
  "comments": "Unknown Type: string,null",
  "x": 0,
  "y": 0,
  "xyAccuracy": "ACC_10KM",
  "country": "AT",
  "canton": "string",
  "district": "Unknown Type: string,null",
  "municipality": "string",
  "fieldName": "string",
  "huntingArea": "Unknown Type: string,null",
  "sensitive": true
}
```

In the attached screenshot, you can see the generated schema ui element. As you can see, enums, nativeEnums and optionals are broken.

![image](https://github.com/blomqma/next-rest-framework/assets/811500/86ff4375-f20d-4047-b1df-2e818fa3010b)

**Comment by blomqma** [comment]
Thanks, I'll look into this soonish.

**Comment by blomqma** [comment]
This is now fixed with latest version v1.2.2.
